### PR TITLE
Update Black tests

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Pull in the latest black snapshot tests

## Test Plan

`cargo test`. I reviewed the new snapshot tests

### CODE REVIEW

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__type_param_defaults.py.snap [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+@@ -8,20 +8,20 @@
+ type D[
+     *something_that_is_very_very_very_long = *something_that_is_very_very_very_long
+ ] = float
+-type something_that_is_long[
+-    something_that_is_long = something_that_is_long
+-] = something_that_is_long
++type something_that_is_long[something_that_is_long = something_that_is_long] = (
++    something_that_is_long
++)
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__type_param_defaults.py.snap:43]: I prefer Ruff's formatting here which is more in line with other assignment formatting where the preference is to break the right side before breaking the left.

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__type_param_defaults.py.snap [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+ 
+ 
+-def simple[
+-    T = something_that_is_long
+-](short1: int, short2: str, short3: bytes) -> float:
++def simple[T = something_that_is_long](
++    short1: int, short2: str, short3: bytes
++) -> float:
+     pass
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__type_param_defaults.py.snap:52]: I prefer Ruff's formatting here. Breaking the return type and arguments is more readable than breaking the type arguments (overall, prefer breaking the right before breaking the left)

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__pep_701.py.snap [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+
+```diff
+--- Black
++++ Ruff
+@@ -119,7 +119,7 @@
+ )
+ 
+ f"{1:{f'{2}'}}"
+-f"{1:{f'{2}'}}"
++f'{1:{f'{2}'}}'
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__pep_701.py.snap:156]: Related to https://github.com/astral-sh/ruff/issues/13237

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__pattern_matching_with_if_stmt.py.snap [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+--- Black
++++ Ruff
+@@ -3,34 +3,36 @@
+         pass
+ 
+ match smth:
+-    case "test" if (
+-        "any long condition" != "another long condition" and "this is a long condition"
+-    ):
++    case "test" if "any long condition" != "another long condition" and "this is a long condition":
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__pattern_matching_with_if_stmt.py.snap:54]: Related to new preview style that parenthesizes long if conditions.

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__function_trailing_comma.py.snap [hunk 5]
```diff
@@ -0,0 +1,569 @@
...
+ 
+ 
+ def foo(
+-    arg: (  # comment with non-return annotation
+-        int
+-        | range
+-        | memoryview
+-        # comment with non-return annotation
+-    ),
++    # comment with non-return annotation
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__function_trailing_comma.py.snap:167]: https://github.com/astral-sh/ruff/issues/13369

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__form_feeds.py.snap [hunk 6]
```diff
@@ -167,8 +167,9 @@ a = [
 +
  #
 -
-+
- #
+-#
+
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__form_feeds.py.snap:171]: Ruff does not support form feed characters.

@konstin I think you were right on the on-site. It's form feed characters :)

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__dummy_implementations.py.snap [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+## Black Differences
+
+```diff
+--- Black
++++ Ruff
+@@ -82,14 +82,12 @@
+ 
+ class ClassC:
+     def f(self):
+-
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__dummy_implementations.py.snap:148]: Ruff does not preserve empty lines at the beginning of a class (intentional decision)

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__comment_after_escaped_newline.py.snap [hunk 8]
```diff
@@ -21,16 +21,16 @@ def bobtwo(): \
...
 +def bob():
 +    # pylint: disable=W9016
      pass
  
  
--def bobtwo():  # some comment here
-+def bobtwo():
-+    # some comment here
+ def bobtwo():
+-
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__comment_after_escaped_newline.py.snap:32]: Ruff does not preserve empty lines between the class and any leading comment (intentional decision)

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__backslash_before_indent.py.snap [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+## Black Differences
+
+```diff
+--- Black
++++ Ruff
+@@ -1,5 +1,4 @@
+ class Plotter:
+-
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__backslash_before_indent.py.snap:27]: Ruff does not preserve empty lines after a class (intentional deviation)

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__form_feeds.py.snap (hunk 6 above)
Comment (konstin) [reply crates/ruff_python_formatter/tests/snapshots/black_compatibility@cases__form_feeds.py.snap:171]: i thought we would, but maybe we only added this for the linter? https://github.com/astral-sh/ruff/pull/7489

**Pull request merged**